tests/conftest.py

```diff
@@ -210,6 +211,17 @@ def test_session_tmp_path(test_fc_session_root_path):
     shutil.rmtree(tmp_path)
 
 
+@pytest.fixture
+def gather_performance_data(request):
```

Comment:
nit: I am imagining this fixture returning a resource. It should be `None` if the flag for dumping the test results was not used. If the flag is used, it can be an object (lets call it `Dumper`), which will be concerned with the root destination of tests dumps. Furthermore, the object should provide interface for setting specific test dump destination under the root destination. In the end, it should provide interface for writing to destination. I would keep the framework agnostic of this tests dump aspect (having it aware only for pass criteria check), delegating the dumping aspect to the test logic. Let me know if you find significant value in this suggestion.